pkg/datagatherer/k8s/generic_test.go

```diff
@@ -0,0 +1,110 @@
+package k8s
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/fake"
+	"k8s.io/utils/diff"
+)
+
+func getObject(version, kind, name, namespace string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": version,
+			"kind":       kind,
+			"metadata": map[string]interface{}{
+				"name": name,
+				"namespace": namespace,
+			},
+		},
+	}
+}
+
+func asUnstructuredList(items ...*unstructured.Unstructured) *unstructured.UnstructuredList {
+	itemsNonPtr := make([]unstructured.Unstructured, len(items))
+	for i, u := range items {
+		itemsNonPtr[i] = *u
+	}
+	return &unstructured.UnstructuredList{
+		Object: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"resourceVersion": "",
+			},
+		},
+		Items: itemsNonPtr,
+	}
+}
+
+func TestGenericGatherer_Fetch(t *testing.T) {
+	emptyScheme := runtime.NewScheme()
```

Comment:
Used an empty scheme here to try and demonstrate it doesn't require anything to be registered/recognised in order to work 😄